### PR TITLE
Small delight tweaks to improve rough corners in the app

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/9251
Fixes https://github.com/vector-im/element-web/issues/14870
Fixes https://github.com/vector-im/element-web/issues/12641

+ Fix left panel layout being wrong when filtering with 0 rooms
+ Fix room list join rooms prompt not showing up due to missing updates
+ Pass filter text when clicking explore/dm prompt
+ Iterate the filtering prompt
image

+ Update Room List filter copy
image

+ Tweak registration based on user feedback
image
image

### CODE REVIEW

**Review by niquewoodhouse — APPROVED** [review]
all looks really good, thanks, we'll iterate after more testing/as part of other issues

File: src/components/views/dialogs/InviteDialog.js [hunk 1]
```diff
@@ -687,123 +697,127 @@ export default class InviteDialog extends React.PureComponent {
...
+                    // We weren't able to find anyone - we're already suggesting the plain email
+                    // as an alternative, so do nothing.
+                    return;
                 }
+
+                // We append the user suggestion to give the user an option to click
+                // the email anyways, and so we don't cause things to jump around. In
+                // theory, the user would see the user pop up and think "ah yes, that
+                // person!"
+                const profile = await MatrixClientPeg.get().getProfileInfo(lookup.mxid);
```
Comment (dbkr) [inline src/components/views/dialogs/InviteDialog.js:789]: You may want to discard results that arrive out of order - as-you-type searching can be a bit subtle to get right: there should be code to crib from elsewhere (we should probably make it a utility class or something).

**Review by dbkr — COMMENTED** [review]
otherwise lgtm

File: src/components/views/dialogs/InviteDialog.js (hunk 1 above)
Comment (t3chguy) [reply src/components/views/dialogs/InviteDialog.js:789]: isn't that on the line below??

Comment (dbkr) [reply src/components/views/dialogs/InviteDialog.js:789]: ah yep, that should do it!

**Review by dbkr — APPROVED** [review]

**Pull request merged**